tests/test_api.py

```diff
@@ -72,41 +73,81 @@ def setUpClass(cls):
     def test_login(self):
         resp = self.app.get('/login')
         self.assertEqual(resp.status_code, 301)
+        self.assertEqual(resp.headers['Location'], '/authorize')
 
     def test_authorize(self):
-        scopes = ["openid", "email", "offline_access"]  # Is offline_access needed for CLI
+        scopes = "openid email offline_access"  # Is offline_access needed for CLI
         CLIENT_ID = "qtMgNk9fqVeclLtZl6WkbdJ59dP3WeAt"
-        REDIRECT_URIS = [
-            "urn:ietf:wg:oauth:2.0:oob",
-            "http://localhost:8080"
-        ]
+        REDIRECT_URI = "http://localhost:8080"
 
         from uuid import uuid4
         state = str(uuid4())
+        query_params = {
+            "response_type": "code",
+            "state": state,
+            "redirect_uri": REDIRECT_URI,
+            "scope": scopes
```

Comment:
missing duration here?

Reply:
it's not actually used or supported by the authorization endpoint